seedfarmer/resources/toolchain_role.template

```diff
@@ -25,7 +25,8 @@ Resources:
               - sts:GetSessionToken
               Effect: Allow
               Resource:
-                Fn::Sub: "arn:${AWS::Partition}:iam::*:role/seedfarmer-{{ project_name }}*"
+                - Fn::Sub: "arn:${AWS::Partition}:iam::*:role/seedfarmer-{{ project_name }}*"
+                - Fn::Sub: "arn:${AWS::Partition}:iam::*:role/*/seedfarmer-{{ project_name }}*"
```

Comment:
Would it make more sense to use the prefix here for least privilege?

```suggestion
                - Fn::Sub: "arn:${AWS::Partition}:iam::*:role{{ role_prefix }}seedfarmer-{{ project_name }}*"
```

Same goes for `deployment_role.template`.

Reply:
This is what I've started with but prefixes used with toolchain, deployment, and module deployment roles can be different (with the current design). We can set it to one specific prefix here but that would essentially mean all prefixes must be the same across all roles (and more importantly all accounts, that may be owned by different teams with different requirements / SCP)

Reply:
Might make sense to leave a comment or something to that?